whylog/config/investigation_plan.py

```diff
@@ -64,22 +59,22 @@ class Clue(object):
     Also, contains parsed line and its source.
     """
 
-    def __init__(self, regex_parameters, line_prefix_content, line_source):
+    def __init__(self, regex_parameters, line_prefix_content, line_offset, line_source):
         self.regex_parameters = regex_parameters
         self.line_prefix_content = line_prefix_content
+        self.line_offset = line_offset
         self.line_source = line_source
 
     def __repr__(self):
-        return "(Clue: %s, %s, %s)" % (
-            self.regex_parameters, self.line_prefix_content, self.line_source
+        return "(Clue: %s, %s, %s, %s)" % (
+            self.regex_parameters, self.line_prefix_content, self.line_offset, self.line_source
         )
 
 
 class LineSource(object):
-    def __init__(self, host, path, offset):
+    def __init__(self, host, path):
         self.host = host
         self.path = path
-        self.offset = offset
```

Comment:
I want decrease number of creating LineSource objects. When LineSource contains offset, Per every matched line
we create LineSource object. Without offset we can creating LineSource per single parsed log file. Offset was moved to Clue class, which is created per every matched parser. Please see my changes in searchers.py.
Next I want move line source to front module and FrontInput will be aggregate object of line source.
I can do it in next PR or even in this. Now I see that @andrzejgorski has this problem in PR #86